src/containers/EventCreationPage.tsx

```diff
@@ -0,0 +1,163 @@
+import Map from "../components/EventCreationPage/Map";
+import NextButton from "../components/EventCreationPage/NextButton";
+import BackButton from "../components/EventCreationPage/BackButton";
+import FormField from "../components/EventCreationPage/FormField";
+import Stepper from "../components/EventCreationPage/Stepper";
+import SelectDateModal from "../components/EventCreationPage/SelectDateModal";
+import { useMutation } from "react-apollo";
+import { ADD_EVENT } from "../graphql/mutations/templates/events";
+import { useEventMutation } from "../graphql/mutations/hooks/events";
+import { isObjectType } from "graphql";
+
+const EventCreationPage = () => {
+  const [openCancelModal, setOpenHandleModal] = useState(false);
+  const [openDateModal, setOpenDateModal] = useState(false);
+  const [complete, setComplete] = useState(false);
+
+  const [eventName, setEventName] = useState<string>("");
+  const [eventDate, setEventDate] = useState<Date | null>(null);
+  const [eventLocation, setEventLocation] = useState<string>("");
+
+  const [activeStep, setActiveStep] = useState<number>(0);
+
+  const handleOpenCancelModal = () => setOpenHandleModal(true);
+  const handleCloseCancelModal = () => setOpenHandleModal(false);
+
+  const handleOpenDateModal = () => setOpenDateModal(true);
+  const handleCloseDateModal = () => setOpenDateModal(false);
+
+  const handleNameChange = (e: any) => {
+    setEventName(e.target.value);
+  };
+  const handleDateChange = (e: any) => {
+    setEventDate(e.target.value);
+  };
+  const handleLocationChange = (e: any) => {
+    setEventLocation(e.target.value);
+  };
+
+  const handleNext = () => {
+    setActiveStep((prevStep) => prevStep + 1);
+  };
+  const handleBack = () => {
+    setActiveStep((prevStep) => prevStep - 1);
+  };
+
+  useEventMutation(
+    {
+      name: eventName,
+      eventDate: eventDate || new Date(),
+      createdBy: 1,
+      isActive: true,
+    },
+    complete,
+    setComplete
+  );
+
+  const handleComplete = () => {
+    setComplete(true);
+    window.location.href = "/events/";
+  };
+
+  const dateParts: {
+    year?: string;
+    month?: string;
+    day?: string;
+    literal?: string;
+  } = eventDate
+    ? new Intl.DateTimeFormat().formatToParts(eventDate).reduce(
+        (obj, currentPart) => ({
+          ...obj,
+          [currentPart.type]: currentPart.value,
+        }),
+        {}
+      )
+    : {};
+  const content =
+    activeStep === 0 ? (
+      <form>
+        <FormField
+          label="Event Name:"
+          placeholder="Event Name Here"
+          onChange={handleNameChange}
+          value={eventName}
+        />
+        <FormField
+          label="Date of Event:"
+          placeholder="YYYY:MM:DD"
+          onChange={handleDateChange}
+          value={
+            eventDate
+              ? `${dateParts.year}:${dateParts.month}:${dateParts.day}`
+              : ""
+          }
+          handleClick={handleOpenDateModal}
```

Comment:
This might be a design question, but the date picker modal only opens on click. Let's say the user enters these field via tabbing from the previous field on the form, would we want the date picker modal to open then too? Consider passing a handler for the `onfocus` event.